rocket-chat-android-widgets/src/main/java/chat/rocket/android/widget/message/MessageFormLayout.java

```diff
@@ -110,17 +101,18 @@ public void afterTextChanged(Editable s) {
     addView(composer);
   }
 
-  public void addExtraActionItem(MessageExtraActionItemPresenter itemPresenter) {
-    boolean found = false;
-    for (MessageExtraActionItemPresenter item : extraActionItems) {
-      if (item.getItemId() == itemPresenter.getItemId()) {
-        found = true;
-        break;
-      }
-    }
+  public void setShowExtraActionSelectionCallback(
+      ShowExtraActionSelectionCallback showExtraActionSelectionCallback) {
+    this.showExtraActionSelectionCallback = showExtraActionSelectionCallback;
+  }
 
-    if (!found) {
-      extraActionItems.add(itemPresenter);
+  public void setTextListener(TextListener textListener) {
+    this.textListener = textListener;
+  }
+
+  private void showExtraActionSelectionDialog() {
```

Comment:
If MessageFormLayout doesn't show Dialog directly, this class should not know that the callback is used for showing dialog. So it is not better to include `dialog` in the method and callback interface.

I think it is better to name this method and the callback interface simply like `OnExtraActionSelectionButtonClicked` and `setOnExtraActionSelectionButtonClickedListener` . (too long 😅 ... more shorter one is prefered :) )